fuse_core/src/async_motion_model.cpp

```diff
@@ -97,6 +105,15 @@ void AsyncMotionModel::initialize(const std::string& name)
   onInit();
 
   executor_->add_node(node_);
+
+  // TODO(CH3): Remove this if the internal node is removed
+  spinning_ = true;
+  spinner_ = std::thread([&](){
+    auto context = node_->get_node_base_interface()->get_context();
+    while(context->is_valid() && spinning_) {
```

Comment:
Executors are already supposed to stop spinning when the context becomes invalid (see [multithreaded executor](https://github.com/ros2/rclcpp/blob/91bc312190e61c8915b0152e5c927c987fc22d6c/rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp#L89)).

I don't think the lambda or loop is needed. The thread could run `executor->spin()`, and `spinning_` being set to `False` could be replaced with `executor->cancel()`.

Reply:
We need the lambda for starting a separate thread to host the blocking spin call (because otherwise the optimizer thread that constructs the motion models and calls initialize() will get blocked).

I'll make use of `cancel()` and remove the `spinning_` flag though!

Reply:
https://github.com/locusrobotics/fuse/pull/286/commits/d9493d49276c9dcab014a65aed5ffbf15aa7d714